data_structures/queue_using_stack.c

```diff
@@ -0,0 +1,89 @@
+/* C Program to implement a queue using two stacks */
+#include <stdio.h> 
+#include <stdlib.h> 
+/* structure of a stack node */
+struct sNode 
+{ 
+    int data; 
+    struct sNode* next; 
+}; 
+void push(struct sNode** top_ref, int new_data); 
+int pop(struct sNode** top_ref);   
+/* structure of queue having two stacks */
+struct queue 
+{ 
+    struct sNode* stack1; 
+    struct sNode* stack2; 
+}; 
+void enQueue(struct queue* q, int x) 
+{ 
+    push(&q->stack1, x); 
```

Comment:
Please catch the case ```q = NULL```. Because otherwise result is ```Segmentation fault```. Your push-function don't catch this case.